Rework migrate data docs (Take 2) (#5244)
elastic/docs-content: manage-data/migrate/migrate-data-between-elasticsearch-clusters-with-minimal-downtime.md

---

> navigation_title: Migrate data using snapshots

Reviewer: with the benefit of hindsight, I wonder if this would just do better to link more deeply into the tutorial ([here](https://docs-v3-preview.elastic.dev/elastic/docs-content/pull/5244/manage-data/migrate/migrate-data-between-elasticsearch-clusters-with-minimal-downtime#migrate-data-snapshot-restore))

  - id: cloud-kubernetes
---

# Migrate {{es}} data using snapshots [migrate-elasticsearch-data-using-snapshots]

> This page explains how to migrate data between {{es}} clusters using incremental snapshots, including an overview, prerequisites, and a recommended timeline for the procedure. When you're ready to begin, refer to [Migrate your data using snapshot and restore](#migrate-data-snapshot-restore) for step-by-step instructions.

Reviewer: consider

```suggestion
This page explains how to migrate data between {{es}} clusters using snapshots. To reduce cluster downtime, Elastic recommends that you use incremental snapshots. This page details the incremental snapshot process, including an overview, prerequisites, and a recommended timeline for the procedure.

:::{tip}
If you don't need to restore incrementally, you can skip to [Migrate your data using snapshot and restore](#migrate-data-snapshot-restore) for basic snapshot and restore instructions.
:::
```

* Maintain data consistency and minimize disruption.  

Snapshot and restore is a reliable approach to migrating any type of {{es}} data. For more detail about the feature, check [Snapshot and restore](/deploy-manage/tools/snapshot-and-restore.md). To learn about other available migration methods and the data types that they support, refer to [](/manage-data/migrate.md).

## How incremental snapshots work [how-incremental-snapshots-work]
Incremental snapshots save only the data that has changed since the last snapshot. The first snapshot is a full copy of the data. Each subsequent snapshot contains only the differences, which makes creating and restoring snapshots faster and more efficient over time. 

When restoring, {{es}} copies only the missing data segments from the snapshot repository to the new cluster local storage. When the changes between snapshots are small, the restore process is significantly faster. 

By taking and restoring incremental snapshots in sequence, you can keep a new cluster closely synchronized with the old cluster, allowing you to migrate most of your data ahead of time and minimize downtime during the final cutover. 

> For more information about migrating your data with snapshot and restore, check [Migrate your data using snapshot and restore](#migrate-data-snapshot-restore).

Reviewer: I think maybe this line is not helpful here

```suggestion
```

### Prerequisites
* Learn how to [set up and manage snapshot repositories](/deploy-manage/tools/snapshot-and-restore/manage-snapshot-repositories.md). 
* If restoring to a different cluster, review [Restore to a different cluster](/deploy-manage/tools/snapshot-and-restore/restore-snapshot.md#restore-different-cluster).
* As an alternative migration method, you can [reindex from a remote cluster](/manage-data/migrate/migrate-data-using-reindex-api.md).

### Requirements

> * **Cluster size** – The new cluster must be the same size or larger than the old cluster. 

Reviewer: listing all of these as requirements feels weird. some of those feel like "considerations" (I realize this predated you)

Author: I've just renamed it to "Requirements and considerations".

    4. Change ingestion and querying to the new cluster. 
    5. Open the indices in the new cluster. 

> ## Migrate your data using snapshot and restore [migrate-data-snapshot-restore]

Author: I've renamed and rearranged the page a bit so that the minimal downtime is still emphasized, and we have a link to the procedure in the intro. Hopefully that makes things more clear and consistent, given that it's a mix of content types. They do fit together nicely (e.g., the suggested timeline), so I didn't want to split into multiple pages.

Follow these steps to migrate your {{es}} data.

### Step 1: Set up the repository in the new cluster [migrate-repo-setup]

In this step, you’ll configure a read-only snapshot repository in the new cluster that points to the storage location used by the old cluster. This allows the new cluster to access and restore snapshots created in the original environment.

::::{tip}

> If your new deployment cannot connect to the same repository used by your self-managed cluster, for example if it's a private Network File System (NFS) share, consider one of the following alternatives:

Reviewer: we don't know if it's a self-managed cluster. consider

```suggestion
If your new deployment cannot connect to the same repository used by your old cluster, for example if it's a self-managed cluster using a private Network File System (NFS) share, consider one of the following alternatives:
```

In this step, you’ll configure a read-only snapshot repository in the new cluster that points to the storage location used by the old cluster. This allows the new cluster to access and restore snapshots created in the original environment.

::::{tip}

> If your new deployment cannot connect to the same repository used by your old cluster, fself-managed cluster, for example if it's a self-managed cluster using a private Network File System (NFS) share, consider one of the following alternatives:

Reviewer: ```suggestion
If your new deployment cannot connect to the same repository used by your old cluster, for example if it's a self-managed cluster using a private Network File System (NFS) share, consider one of the following alternatives:
```
